### PR TITLE
boards: added mikroelektronika esp32 wifi shield

### PR DESCRIPTION
**Pull request by uLipe**
Adds support for the esp32 based MikroElektronika shields to be used as AT wifi driver.

### CODE REVIEW

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+the standard MikroBus interface with UART pins plus its Espressif AT
+command set. Also Zephyr modem driver could be used in place, to used
+this module as a native WIFI networking interface
+
+.. image:: ./wifi_click.jpeg
+   :width: 634px
+   :align: center
+   :alt: ESP-01
+
+Pins Assignment of the ESP-01 Module
```
Comment (nandojve) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:20]: I think ESP-01 only works with CONFIG_WIFI_ESP_AT_VERSION_1_7 because of flash size, at least that is the last info that I have. Is espressif released a 2.0 version for ESP-01 that fits on ESP-01?

**Review by nandojve — CHANGES_REQUESTED** [review]

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst (hunk 1 above)
Comment (uLipe) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst:20]: Another copy pasta.

File: boards/shields/esp_8266/Kconfig.defconfig [hunk 2]
```diff
@@ -12,7 +12,7 @@ config WIFI_ESP_AT
 	depends on WIFI
 
 choice WIFI_ESP_AT_VERSION
-	default WIFI_ESP_AT_VERSION_1_7
+	default WIFI_ESP_AT_VERSION_2_0
```
[the remark this answers is not in the document]
Comment (uLipe) [reply boards/shields/esp_8266/Kconfig.defconfig]: Up to date.

**Comment by uLipe** [comment]
@nandojve PTAL again.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+   :align: center
+   :alt: MikroE WIFI BLE Click
+
+Pins Assignment of the Shield
+=============================
+
++-----------------------+---------------------------------------------+
+| Shield Connector Pin  | Function                                    |
++=======================+=============================================+
+| RST#                  | ESP32 Reset                                 |
```
Comment (microbuilder) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:26]: We should add a note that there is a conflict with the rst polarity and we need to disconnect the RST pin on most board (the LPC55S69 at least).

**Comment by uLipe** [comment]
@microbuilder PTAL again as well.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+This shield should be loaded with the `ESP32 AT Bin`_ software which is
+available at Espressif Systems web site. This version is command
+compatible with ESP32 AT Bin 2.0.0
+
+.. note::
+   Boards that already have a network interface:  Check the network
+	documentation to understand how to properly configure both interfaces.
+	To keep things simple, make sure WiFi is the only interface enabled in
+	Networking -> Link Layer Options.  This will avoid problems running
+	Zephyr samples.
```
Comment (microbuilder) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:54]: Whitespace

**Review by microbuilder — CHANGES_REQUESTED** [review]

**Comment by uLipe** [comment]
@nandojve , @microbuilder  PTAL again.

**Review by nandojve — DISMISSED** [review]
LGTM!

Tip: Try use double arrow at reviewers name. That tell us when you sent a fix that we requested!

**Comment by mmahadevan108** [comment]
Thank you for the contribution. How was this tested?

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+UART interface. (see :ref:`shields` for more details).
+
+.. note::
+   Sometimes boards declare standard headers like Arduino R3 but don't
+   define all connections.  Make sure that the board you are using includes
+   all required definitions to avoid build errors.
+
+This shield should be loaded with the `ESP32 AT Bin`_ software which is
+available at Espressif Systems web site. This version is command
+compatible with ESP32 AT Bin 2.0.0
```
Comment (microbuilder) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:47]: Sorry to nag, but could you perhaps add a comment and code block here showing how people can flash the said firmware.

It takes a bit of digging to find the right command, and it's important to be able to reproduce the same device config used by this PR:

```bash
$ esptool.py \
  --chip auto \
  --baud 115200 \
  --before default_reset \
  --after hard_reset write_flash \
  --flash_mode dio \
  --flash_freq 40m \
  --flash_size detect \
  0x10000 ota_data_initial.bin \
  0x1000 bootloader/bootloader.bin \
  0x20000 at_customize.bin \
  0x21000 customized_partitions/ble_data.bin \
  0x24000 customized_partitions/server_cert.bin \
  0x26000 customized_partitions/server_key.bin \
  0x28000 customized_partitions/server_ca.bin \
  0x2a000 customized_partitions/client_cert.bin \
  0x2c000 customized_partitions/client_key.bin \
  0x2e000 customized_partitions/client_ca.bin \
  0x30000 customized_partitions/factory_param.bin \
  0xf000 phy_init_data.bin 0x100000 esp-at.bin \
  0x8000 partitions_at.bin
```

**Review by microbuilder — CHANGES_REQUESTED** [review]

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+   Boards that already have a network interface:  Check the network
+   documentation to understand how to properly configure both interfaces.
+   To keep things simple, make sure WiFi is the only interface enabled in
+   Networking -> Link Layer Options.  This will avoid problems running
+   Zephyr samples.
+
+Build and Programming
+*********************
+
+Set ``-DSHIELD=<mikroe_wiffi_bt>`` when you invoke ``west build``.
```
Comment (microbuilder) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:59]:
```suggestion
Set ``-DSHIELD=<mikroe_wifi_bt>`` when you invoke ``west build``.
```

**Review by microbuilder — CHANGES_REQUESTED** [review]

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+To build shield with specific overlay:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/wifi
+   :board: lpcxpresso55s69_cpu0
+   :shield: mikroe_wifi_bt
```
Comment (microbuilder) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:66]: Should this be:

```suggestion
   :shield: mikroe_wifi_bt_click
```

???

**Review by microbuilder — CHANGES_REQUESTED** [review]

**Comment by microbuilder** [comment]
> Thank you for the contribution. How was this tested?

With the lpcxpresso55s69 and the Mikroe Wifi BLE Click board, which has an ESP-WROOM-32 modules on it, reflashed with the 2.0.0 ESP32 AT firmware as mentioned in the guide. The GND, 3.3V, TXD and RXD pins were connected to the appropriate pins on the LPC55S69. The sample used was `samples/net/wifi`.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst (hunk 5 above)
Comment (microbuilder) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst:47]: This is also missing the serial port, baud rate, etc., which should be added just for a working example even if it's a bit OS-specific in the case of the serial port.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst (hunk 6 above)
Comment (microbuilder) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst:59]: Spelling (`wiffi`), but is the shield name `mikroe_wifi_bt` or `mikroe_wifi_bt_click`?

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst (hunk 5 above)
Comment (uLipe) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst]: It is possible to use command without those parameters, esptool will search the board automagically, altough it is possible to supply port and baud-rate, I added these section in the docs.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst (hunk 7 above)
Comment (uLipe) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst]: It seems I left this way while I was deciding the name. :D

**Comment by uLipe** [comment]
@mmahadevan108 besides the steps mentioned by @microbuilder, the test can be performed by just building the `sample/net/wifi` for a board with this shield wired.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+Alternatively, it is possible to select the port and the baud rate:
+
+.. code-block:: console
+esptool.py --chip auto --port <Selected PORT>  --baud 115200 --before default_reset --after hard_reset write_flash --flash_mode dio --flash_freq 40m --flash_size detect 0x10000 ota_data_initial.bin 0x1000 bootloader/bootloader.bin 0x20000 at_customize.bin 0x21000 customized_partitions/ble_data.bin 0x24000 customized_partitions/server_cert.bin 0x26000 customized_partitions/server_key.bin 0x28000 customized_partitions/server_ca.bin 0x2a000 customized_partitions/client_cert.bin 0x2c000 customized_partitions/client_key.bin 0x2e000 customized_partitions/client_ca.bin 0x30000 customized_partitions/factory_param.bin 0xf000 phy_init_data.bin 0x100000 esp-at.bin 0x8000 partitions_at.bin
+
+After flashed, it is possible to verify the module, while connnected, open your preferred terminal configured as 115200, 8, N, 1 and
+perform a board reset, you should see an initial log and it last message should be the version of the AT firmware flashed.
```
Comment (erwango) [inline boards/shields/mikroe_wifi_bt_click/doc/index.rst:62]:
```suggestion
Once flashed, it is possible to verify the module. While connected, open your preferred
terminal configured as 115200, 8, N, 1 and perform a board reset. You should see an
initial log and last message should be the version of the AT firmware flashed.
```

File: drivers/wifi/esp_at/esp.h [hunk 9]
```diff
@@ -109,8 +109,10 @@ extern "C" {
 
 #if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
 #define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
+#elif defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
+#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
 #else
-#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))",0"
+#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))" ,0"
```
Comment (erwango) [inline drivers/wifi/esp_at/esp.h:115]: Please split this in a dedicated commit with a proper description.
It could be useful for instance if one would like to revert this change while keeping the shield definition.

**Review by erwango — CHANGES_REQUESTED** [review]
Minor remarks, also please update `samples/net/wifi/sample.yaml` and add this shield in order to get it built in CI.

File: drivers/wifi/esp_at/esp.h (hunk 9 above)
Comment (uLipe) [reply drivers/wifi/esp_at/esp.h:115]: Added a commit for it.

**Review by microbuilder — DISMISSED** [review]

**Review by nandojve — DISMISSED** [review]

**Comment by uLipe** [comment]
Folks, just added another example of devkit with this one enabled and connected through an arduino connector standard UART pins.

**Review by microbuilder — DISMISSED** [review]

**Comment by microbuilder** [comment]
@erwango Mind having another look to try to get this in before the freeze? Thanks!

**Review by nandojve — DISMISSED** [review]

File: boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi [hunk 10]
```diff
@@ -146,3 +153,6 @@ zephyr_udc0: &usbotg_fs {
 &iwdg {
 	status = "okay";
 };
+
+mikrobus_serial: &usart3 {
+};
```
Comment (erwango) [inline boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi:158]: You can't do this. Unless I've missed it, there's no actual mikrobus connector on this board

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi (hunk 10 above)
Comment (microbuilder) [reply boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi:158]: You're right. Ideally there should be a `arduino_usart` definition for this board, in addition to the existing `arduino_i2c` and `arduino_spi`, and we assign the module to that.

There are dedicate UART pins on the Uno header, but it doesn't seem very common in Zephyr to configure them. This is all I could find:

```bash
boards/arm/b_l072z_lrwan1/b_l072z_lrwan1.dts
97:arduino_usart: &usart1 {};
```

Comment (microbuilder) [reply boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi:158]: @erwango Should we add `arduino_usart` in here: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/b_u585i_iot02a/arduino_r3_connector.dtsi

Comment (erwango) [reply boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi:158]: Yes, it would work. It's not common because not frequently needed, but it's valid.

File: boards/shields/mikroe_wifi_bt_click/boards/lpcxpresso55s69_cpu0.overlay [hunk 11]
```diff
@@ -0,0 +1,7 @@
+/* Copyright (c) 2022 Linaro Limited.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&mikrobus_serial {
+	status = "okay";
+};
```
Comment (erwango) [inline boards/shields/mikroe_wifi_bt_click/boards/lpcxpresso55s69_cpu0.overlay:7]: I think this shouldn't be required, as this is part of mikroe_wifi_bt_click.overlay

File: boards/shields/mikroe_wifi_bt_click/boards/lpcxpresso55s69_ns.overlay [hunk 12]
```diff
@@ -0,0 +1,7 @@
+/* Copyright (c) 2022 Linaro Limited.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&mikrobus_serial {
+	status = "okay";
+};
```
Comment (erwango) [inline boards/shields/mikroe_wifi_bt_click/boards/lpcxpresso55s69_ns.overlay:7]: ditto

File: boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi (hunk 10 above)
Comment (erwango) [reply boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi:158]: What will not work is to build mikroe shield with this board though.
Building with `-DSHIELD=mikroe_wifi_bt_click` will get `mikroe_wifi_bt_click.overlay` in dts, which can't build with a board not defining `mikrobus_serial`.
Unfortunately device tree connectors description is too much close to hw reality: if not fitting, it doesn't compile. And to get it work you need to use wires on hw and hack dts...
But unfortunately, I don't have solution to overcome this with a solution acceptable upstream

Comment (erwango) [reply boards/arm/b_u585i_iot02a/b_u585i_iot02a-common.dtsi]: Ok, I forgot about the possibility to add other overlays. That would work indeed.

**Review by mniestroj — CHANGES_REQUESTED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
@uLipe This is getting a bit messy. There are removals of additions in unrelated commits. That's getting hard to review. Would you mind doing some clean up in the commits ?
Once you're settled on the actual changes to be kept, I'd suggest to squash everything and then split the resulting commit into coherent individual commits.

File: drivers/wifi/esp_at/esp.h [hunk 13]
```diff
@@ -109,6 +109,8 @@ extern "C" {
 
 #if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
 #define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
+#elif defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
+#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
 #else
 #define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))" ,0"
 #endif
```
[the remark this answers is not in the document]
Comment (uLipe) [reply drivers/wifi/esp_at/esp.h]: V2.0 is a stable version and was out there for a while.

I added the option for the V2.1 as well and cleaned up this case, please check if it is better now.

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+   0x24000 customized_partitions/server_cert.bin \
+   0x26000 customized_partitions/server_key.bin \
+   0x28000 customized_partitions/server_ca.bin \
+   0x2a000 customized_partitions/client_cert.bin \
+   0x2c000 customized_partitions/client_key.bin \
+   0x2e000 customized_partitions/client_ca.bin \
+   0x30000 customized_partitions/factory_param.bin \
+   0xf000 phy_init_data.bin \
+   0x100000 esp-at.bin \
+   0x8000 partitions_at.bin
```
[the remark this answers is not in the document]
Comment (microbuilder) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst]: @mniestroj  Can we address this, at worst, in a documentation update to be able to get this in before the feature freeze today? The command above is valid, and is provided to be helpful, but we're free to improve this before the final 3.1 tag as a documentation fix.

**Review by erwango — DISMISSED** [review]
Approved for me, the AT Command version point set aside.

**Review by microbuilder — DISMISSED** [review]

File: drivers/wifi/esp_at/esp.h [hunk 15]
```diff
@@ -107,10 +107,10 @@ extern "C" {
 #define ESP_MODE_AP		2
 #define ESP_MODE_STA_AP		3
 
-#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
-#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
-#else
+#if defined(WIFI_ESP_AT_VERSION_1_7) || defined(WIFI_ESP_AT_VERSION_2_0)
```
Comment (JordanYates) [inline drivers/wifi/esp_at/esp.h:110]: This is incorrect. `AT+CWMODE=<mode>,[autoconnect]` appears to be be the default moving forward:
[v2.1.0 docs](https://docs.espressif.com/projects/esp-at/en/release-v2.1.0.0_esp32/AT_Command_Set/Wi-Fi_AT_Commands.html#at-cwmodesets-the-wi-fi-mode-station-softap-station-softap)
[v2.2.0 docs](https://docs.espressif.com/projects/esp-at/en/release-v2.2.0.0_esp32/AT_Command_Set/Wi-Fi_AT_Commands.html#set-command)
[v2.3.0 docs](https://espressif-docs.readthedocs-hosted.com/projects/esp-at/en/release-v2.3.0.0_esp32c3/AT_Command_Set/Wi-Fi_AT_Commands.html#set-command)
But this is explicitly opting into it for the old versions only.

**Review by nandojve — DISMISSED** [review]

**Review by JordanYates — CHANGES_REQUESTED** [review]

File: drivers/wifi/esp_at/esp.h (hunk 15 above)
Comment (JordanYates) [reply drivers/wifi/esp_at/esp.h:110]: I agree that my original change was technically incorrect because I was testing it against AT version 2.1.0.0 while using `CONFIG_WIFI_ESP_AT_VERSION_2_0` (as higher versions did not exist)

Comment (uLipe) [reply drivers/wifi/esp_at/esp.h]: @JordanYates ,  it appears to be something outdated in firmware when comparing  the documentation, we have flashed a bunch of  these click shields here, and all of them returned `ERROR` after using the CWMODE with two arguments, it seems the V2.0 still uses the legacy CWMODE with single argument.

A addressed the missing `CONFIG_ ` part.

**Review by erwango — DISMISSED** [review]

**Review by microbuilder — APPROVED** [review]

**Review by microbuilder — DISMISSED** [review]

File: drivers/wifi/esp_at/Kconfig.esp_at [hunk 16]
```diff
@@ -144,6 +144,11 @@ config WIFI_ESP_AT_VERSION_2_0
 	help
 	  Use AT command set version 2.0.
 
+config WIFI_ESP_AT_VERSION_2_1
+	bool "AT version 2.1"
+	help
+	  Use AT command set version 2.1.
+
```
Comment (mniestroj) [inline drivers/wifi/esp_at/Kconfig.esp_at:151]: I think we should default to this option, as this is the newest firmware version we support.

File: drivers/wifi/esp_at/esp.h [hunk 17]
```diff
@@ -107,10 +107,10 @@ extern "C" {
 #define ESP_MODE_AP		2
 #define ESP_MODE_STA_AP		3
 
-#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
-#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
-#else
+#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7) || defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
 #define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))",0"
+#else
+#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
```
Comment (mniestroj) [inline drivers/wifi/esp_at/esp.h:113]: There is also
```
#if defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
		SETUP_CMD_NOHANDLE(ESP_CMD_CWMODE(STA)),
		SETUP_CMD_NOHANDLE("AT+CWAUTOCONN=0"),
		SETUP_CMD_NOHANDLE(ESP_CMD_CWMODE(NONE)),
#endif
```
which need to be enabled for 2.1+ firmware versions as well.

**Review by mniestroj — CHANGES_REQUESTED** [review]

**Comment by microbuilder** [comment]
@mniestroj is the current code OK for you? If you can review again, today is the last day to merge this in for inclusion in 3.1.

File: drivers/wifi/esp_at/esp.h (hunk 15 above)
Comment (microbuilder) [reply drivers/wifi/esp_at/esp.h]: @JordanYates Any feedback?

**Review by mniestroj — DISMISSED** [review]
LGTM!

**Review by microbuilder — DISMISSED** [review]

File: boards/shields/mikroe_wifi_bt_click/doc/index.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (mniestroj) [reply boards/shields/mikroe_wifi_bt_click/doc/index.rst]: Sure, that is not a blocker from my side.

File: drivers/wifi/esp_at/esp.h (hunk 15 above)
Comment (JordanYates) [reply drivers/wifi/esp_at/esp.h]: > it seems the V2.0 still uses the legacy CWMODE with single argument

Yes, thats what I was implying. I had `CONFIG_WIFI_ESP_AT_VERSION_2_0` selected while the ESP firmware was running v2.1.0

File: drivers/wifi/esp_at/esp.h [hunk 18]
```diff
@@ -107,10 +107,10 @@ extern "C" {
 #define ESP_MODE_AP		2
 #define ESP_MODE_STA_AP		3
 
-#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
-#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
-#else
+#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7) || defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
```
Comment (JordanYates) [inline drivers/wifi/esp_at/esp.h:110]: Sorry, this is still the wrong way around.
You are using the new command arguments (`AT+CWMODE=<mode>,[autoconnect]`) with the old AT versions.

The diff here should only be changing this:
`#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)`
to this:
`#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7) || defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)`

**Review by JordanYates — CHANGES_REQUESTED** [review]

File: drivers/wifi/esp_at/esp.h (hunk 18 above)
Comment (microbuilder) [reply drivers/wifi/esp_at/esp.h:110]: @JordanYates care to make a specific suggestion in a comment, and I can merge that in to see if we can get this into 3.1?

Comment (uLipe) [reply drivers/wifi/esp_at/esp.h:110]: Sorry about that, I forgot to invert the statements, and when tested I was with the 2.1 version flashed in my board...

Updated

**Comment by uLipe** [comment]
If there any chance to review , I re-tested new and legacy versions.

**Review by JordanYates — CHANGES_REQUESTED** [review]
Looking through all the uses of `CONFIG_WIFI_ESP_AT_VERSION` in the source, I found this:
https://github.com/zephyrproject-rtos/zephyr/blob/bfe16bdc1713f8eed507375f2f3643659b5bbb55/drivers/wifi/esp_at/esp.c#L973-L977
Given thats the location that uses the `ESP_CMD_CWMODE` macro, I suspect it also needs to be changed.

Looking at the commit that added the code https://github.com/zephyrproject-rtos/zephyr/commit/03ce61004b355ff8e8c1cc1fd6700a3b10ea4b18, it looks like this could be replaced with just `SETUP_CMD_NOHANDLE(ESP_CMD_CWMODE(NONE))` on the newer versions? Given that AUTOCONN is now an argument to CWMODE.

File: drivers/wifi/esp_at/esp.c [hunk 19]
```diff
@@ -951,7 +951,7 @@ static void esp_init_work(struct k_work *work)
 		SETUP_CMD_NOHANDLE(
 			ESP_CMD_CWLAPOPT(ESP_CMD_CWLAPOPT_ORDERED, ESP_CMD_CWLAPOPT_MASK)),
 
-#if defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
+#if !defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
```
Comment (uLipe) [inline drivers/wifi/esp_at/esp.c:954]: @JordanYates it was replaced already.

**Comment by uLipe** [comment]
Twister error seems to not be related to this PR.

File: drivers/wifi/esp_at/esp.c (hunk 19 above)
Comment (JordanYates) [reply drivers/wifi/esp_at/esp.c:973]: Whoops, my bad

**Review by JordanYates — DISMISSED** [review]
You'll need to rebase the branch instead of adding a merge commit, but the protocol version changes looks fine to me.

**Comment by uLipe** [comment]
Just rebased to main.

**Review by microbuilder — APPROVED** [review]

**Comment by microbuilder** [comment]
@mniestroj @erwango @carlescufi any chance of a new review to see if we can merge this in before the 3.1 freeze, or is it already too late?

**Review by mniestroj — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]
